apis/server/router.go

```diff
@@ -39,7 +39,7 @@ func initRoute(s *Server) http.Handler {
 	r.Path(versionMatcher + "/containers/{name:.*}/exec").Methods(http.MethodPost).Handler(s.filter(s.createContainerExec))
 	r.Path(versionMatcher + "/exec/{name:.*}/start").Methods(http.MethodPost).Handler(s.filter(s.startContainerExec))
 	r.Path(versionMatcher + "/containers/{id:.*}/rename").Methods(http.MethodPost).Handler(s.filter(s.renameContainer))
-	r.Path("/containers/{id:.*}/restart").Methods(http.MethodPost).Handler(s.filter(s.restartContainer))
+	r.Path(versionMatcher + "/containers/{id:.*}/restart").Methods(http.MethodPost).Handler(s.filter(s.restartContainer))
```

Comment:
Actually this place should be `/containers/{name:.*}/restart` rather than `/containers/{id:.*}/restart`.

You can find code https://github.com/alibaba/pouch/blob/master/apis/server/container_bridge.go#L65. It is finding `name`. @HusterWan 

I will fix this soon.